tests/test_request_message.py

```diff
@@ -0,0 +1,98 @@
+    (1001, 1005, 123456, 400, b'{"Error":"Bad Request.FOREIGN KEY constraint failed"}\n')
+    ))
+def test_sending_request_fail(client, app, sender_id, receiver_id, request_time, status_code, response_message):
+
+    response = client.post(f"/api/request/send/{sender_id}/{receiver_id}/{request_time}")
+    assert response.status_code == status_code
+    assert response.data == response_message
+
+# test accepting request successfully
+def test_accepting_request(client, app):
+    response = client.post("/api/request/accept/1001/123456")
+    assert response.status_code == 200
+    assert response.data == b'Success'
+
+    with app.app_context():
+        assert query_db(
+                "select * from conversation_request where id=:request_id",
+                {"request_id": 1001},
+                one=True
+                )["accepted"] == 1
+
+# test accepting request fail
+def test_accepting_request_fail(client, app):
+    response = client.post("/api/request/accept/1002/123456")
+    assert response.status_code == 400
+    assert response.data == b'{"Error":"Bad Request.No request found"}\n'
+
+# test reject request successfully
+def test_rejecting_request(client, app):
+    response = client.post("/api/request/reject/1001")
+    assert response.status_code == 200
+    assert response.data == b'Success'
+
+    with app.app_context():
+        assert query_db(
+                "select * from conversation_request where id=:request_id",
+                {"request_id": 1001},
+                one=True
+                )["accepted"] == 0
+
+# test rejecting request fail
+def test_rejecting_request_fail(client, app):
+    response = client.post("/api/request/reject/1002")
+    assert response.status_code == 400
+    assert response.data == b'{"Error":"Bad Request.No request found"}\n'
+
+#test get all requests successfully
+@pytest.mark.parametrize(("user_id", "status_code"), (
+    (1000, 200),
+    (1001, 200),
+    (1002, 200)
+    ))
+def test_get_all_requests(client, app, user_id, status_code):
+    response = client.get(f"/api/request/all/{user_id}")
+    assert response.status_code == status_code
+    data_retrieved = json.loads(response.data.decode('utf8'))
+
+    with app.app_context():
+        assert query_db(
+                "select * from conversation_request where receiver_id=:user_id and accepted=0",
+                {"user_id": user_id}
+                ) == data_retrieved
+
+#test get all requests successfully
+@pytest.mark.parametrize(("user_id", "status_code", "response_message"), (
+    (1004, 400, b'{"Error":"Bad Request.No user found"}\n'),
+    (1005, 400, b'{"Error":"Bad Request.No user found"}\n'),
+    (1006, 400, b'{"Error":"Bad Request.No user found"}\n')
+    ))
+def test_get_all_requests_fail(client, app, user_id, status_code, response_message):
+    response = client.get(f"/api/request/all/{user_id}")
+    assert response.status_code == status_code
+    assert response.data == response_message
+
```

Comment:
You should group them into a Python class.

Also, with each assert statement, please add a more explanatory/human readable description of why the test fails (where relevant). For example:
`assert response.data == "some data", "Fetching all conversation requests failed"`